File: app/src/main/java/org/dicio/dicio_android/skills/navigation/NavigationInfo.java
package org.dicio.dicio_android.skills.navigation;

import static org.dicio.dicio_android.SectionsGenerated.navigation;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import org.dicio.dicio_android.R;
import org.dicio.dicio_android.Sections;
import org.dicio.skill.Skill;
import org.dicio.skill.SkillContext;
import org.dicio.skill.SkillInfo;
import org.dicio.skill.chain.ChainSkill;
import org.dicio.skill.standard.StandardRecognizer;

public class NavigationInfo extends SkillInfo {
    public NavigationInfo() {
        super("navigation", R.string.skill_name_navigation,
                R.string.skill_sentence_example_navigation, R.drawable.ic_navigate_white,
                false);
    }

    @Override
    public boolean isAvailable(final SkillContext context) {
        return Sections.isSectionAvailable(navigation);
    }

    @Override
    public Skill build(final SkillContext context) {
        return new ChainSkill.Builder()
                .recognize(new StandardRecognizer(Sections.getSection(navigation)))
                .process(new NavigationProcessor())
                .output(new NavigationOutput());
    }

    @Nullable
    @Override
    public Fragment getPreferenceFragment() {
        return null;
    }
}

File: app/src/main/kotlin/org/stypox/dicio/skills/navigation/NavigationInfo.kt
package org.stypox.dicio.skills.navigation

import android.content.Context
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Directions
import androidx.compose.material.icons.filled.MusicNote
import androidx.compose.material.icons.filled.Navigation
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.vector.rememberVectorPainter
import androidx.fragment.app.Fragment
import org.dicio.skill.skill.Skill
import org.dicio.skill.context.SkillContext
import org.dicio.skill.skill.SkillInfo
import org.stypox.dicio.R
import org.stypox.dicio.Sections
import org.stypox.dicio.SectionsGenerated.navigation
import org.stypox.dicio.sentences.Sentences

object NavigationInfo : SkillInfo("navigation") {
    override fun name(context: Context) =
        context.getString(R.string.skill_name_navigation)

    override fun sentenceExample(context: Context) =
        context.getString(R.string.skill_sentence_example_navigation)

    @Composable
    override fun icon() =
        rememberVectorPainter(Icons.Default.Directions)

    override fun isAvailable(ctx: SkillContext): Boolean {
        return Sentences.Navigation[ctx.locale.language] != null
    }

    override fun build(ctx: SkillContext): Skill<*> {
        return NavigationSkill(NavigationInfo, Sentences.Navigation[ctx.locale.language]!!)
    }
}
